src/cudadecoderbin/batched-wav-nnet3-cuda.cc

```diff
@@ -90,13 +90,14 @@ void FinishOneDecode(const std::string &utt, const std::string &key,
                      int64 *num_frames, double *tot_like,
                      CompactLatticeWriter *clat_writer,
                      std::mutex *clat_writer_mutex, std::mutex *stdout_mutex,
-                     CompactLattice &clat) {
+                     const bool write_lattice, CompactLattice &clat) {
   nvtxRangePushA("FinishOneDecode");
   GetDiagnosticsAndPrintOutput(utt, word_syms, clat, stdout_mutex, num_frames,
                                tot_like);
+  if(write_lattice)
   {
```

Comment:
can you put the { on the same line as the if?  And space after 'if'.
Closing the clat_writer at the end is not needed, the destructor will do it.